### PR TITLE
Fixes #26758 - unify renderer_error helper

### PR DESCRIPTION
**Pull request by lzap**
Preview actually throws argument error (3 vs 1).

### CODE REVIEW

**Comment by lzap** [comment]
Ho ho ho, tests failed. Hmmm.

File: app/controllers/unattended_controller.rb [hunk 1]
```diff
@@ -80,14 +80,6 @@ def preview?
     params.key?(:spoof) || params.key?(:hostname)
   end
 
-  def render_error(status, error_message, params)
```
Comment (timogoebel) [inline app/controllers/unattended_controller.rb:83]: Can we change this to

```ruby
      def render_error(options)
        return super unless ipxe_request?
        render_ipxe_message(message: options[:message], status: options[:status] || :not_found)
      end
```

Comment (lzap) [reply app/controllers/unattended_controller.rb:83]: Sure, probably better idea.

**Comment by lzap** [comment]
Let's see if tests pass.

File: app/controllers/unattended_controller.rb [hunk 2]
```diff
@@ -171,7 +168,15 @@ def verify_found_host
     end
 
     error = host_verifier.errors.first
-    render_error(error[:type], error[:message], error[:params])
+    if ipxe_request?
+      render_ipxe_message(error[:message])
```
Comment (timogoebel) [inline app/controllers/unattended_controller.rb]: I believe this needs to be

```suggestion
      render_ipxe_message(message: error[:message])
```

But shouldn't `render_error` handle this on it's own?

**Comment by lzap** [comment]
Thanks, added your commit as "refs". Let's see.

**Comment by timogoebel** [comment]
Tests are still 🔴 :-(

**Comment by lzap** [comment]
Ok it was a mess, now it follows the same parameters as other controllers have: message and options hash.

File: app/controllers/unattended_controller.rb [hunk 3]
```diff
@@ -171,7 +171,11 @@ def verify_found_host
     end
 
     error = host_verifier.errors.first
-    render_error(error[:type], error[:message], error[:params])
+    if ipxe_request?
```
Comment (timogoebel) [inline app/controllers/unattended_controller.rb]: Doesn't `render_error` already handle this path correctly?

Comment (lzap) [reply app/controllers/unattended_controller.rb]: Hmmm refactoring of refactored code and you end up with this ;-)

**Comment by lzap** [comment]
Rebased, fixed missed test.

File: app/controllers/unattended_controller.rb [hunk 4]
```diff
@@ -80,12 +80,12 @@ def preview?
...
-    logger.error error_message % params
-    message = _(error_message) % params
-    return render_ipxe_message(message: message, status: status) if ipxe_request?
-    # add a comment character (works with Red Hat and Debian systems) to avoid parsing errors
-    render(:plain => "# #{message}", :status => status, :content_type => 'text/plain')
+  def render_error(message, options)
+    if ipxe_request?
+      render_ipxe_message(message: message, status: options[:status] || :not_found)
+    else
+      super(message, options)
```
Comment (timogoebel) [inline app/controllers/unattended_controller.rb]: Nit: You can just call `super` here. No need to pass the parameters.

**Comment by lzap** [comment]
Rebased.

During RC testing I found that actually we have a bug in core which is fixed by this PR:

```
2019-08-20T11:41:56 [W|app|8cd14d86] Action failed
ArgumentError: wrong number of arguments (given 1, expected 3)
/usr/share/foreman/app/controllers/unattended_controller.rb:83:in `render_error'
/usr/share/foreman/app/controllers/concerns/foreman/controller/template_rendering.rb:19:in `rescue in safe_render'
/usr/share/foreman/app/controllers/concerns/foreman/controller/template_rendering.rb:15:in `safe_render'
/usr/share/foreman/app/controllers/concerns/foreman/controller/template_rendering.rb:9:in `render_template'
/usr/share/foreman/app/controllers/unattended_controller.rb:130:in `render_provisioning_template'
```

This appears only when there is a rendering error (which is the case for RC). I will propose this one for cherry pick.

**Comment by tbrisker** [comment]
[test foreman]

**Review by timogoebel — APPROVED** [review]
LGTM.

**Comment by tbrisker** [comment]
1.23 - 11e8580e4

**Pull request merged**